lib/internal/Magento/Framework/File/Transfer/Adapter/Http.php

```diff
@@ -19,13 +24,16 @@ class Http
     private $mime;
 
     /**
+     * @param \Magento\Framework\App\Request\Http $request
      * @param \Magento\Framework\App\Response\Http $response
      * @param \Magento\Framework\File\Mime $mime
      */
     public function __construct(
+        \Magento\Framework\App\Request\Http $request,
         \Magento\Framework\HTTP\PhpEnvironment\Response $response,
         \Magento\Framework\File\Mime $mime
     ) {
+        $this->request = $request;
```

Comment:
1. Please make this argument last one
2. Please make it optional & fallback to object manager to make this change backward compatible

Reply:
Done!